src/main/java/org/spongepowered/api/entity/vehicle/minecart/MinecartLike.java

```diff
@@ -90,4 +85,23 @@
     default Value.Mutable<Vector3d> derailedVelocityModifier() {
         return this.requireValue(Keys.DERAILED_VELOCITY_MODIFIER).asMutable();
     }
+
+    /**
+     * {@link Keys#MINECART_BLOCK_OFFSET}
+     *
+     * @return The represented block's offset of the minecart
+     */
+    default Value.Mutable<Integer> minecartBlockOffset() {
+        return this.requireValue(Keys.MINECART_BLOCK_OFFSET).asMutable();
+    }
+
+    /**
+     * {@link Keys#BLOCK_STATE}
+     *
+     * @return The block state of the block occupied minecart
+     */
+    default Optional<Value.Mutable<BlockState>> block() {
```

Comment:
Why was this moved up to `MinecartLike`?

Reply:
https://github.com/SpongePowered/Sponge/blob/1d33b3c3daf2fbcee44554d128deec16c11aa96e/src/main/java/org/spongepowered/common/data/provider/entity/AbstractMinecartData.java#L50-L54

Reply:
So this is a strange one in Vanilla:

```java
    public BlockState getDisplayBlockState() {
        return !this.hasCustomDisplay() ? this.getDefaultDisplayBlockState() : Block.stateById((Integer)this.getEntityData().get(DATA_ID_DISPLAY_BLOCK));
    }

    public BlockState getDefaultDisplayBlockState() {
        return Blocks.AIR.defaultBlockState();
    }
    ```
    
Just why Mojang...

I think we need to ignore that `AbstractMinecart` has this and keep it on our `BlockOccupiedMinecart`